app/models/meter_reading.rb

```diff
@@ -0,0 +1,43 @@
+class MeterReading < ApplicationRecord
+  VALID_DATE = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/.freeze
+
+  belongs_to :billing_contract, optional: true
+
+  validates :date, presence: true, format: { with: VALID_DATE,
+                                             message: 'is invalid date' }
+  validates :value, presence: true, numericality: { only_integer: true }
+
+  validate :value_bigger_than_previous_one?
+
+  private
+
+  def value_bigger_than_previous_one?
+    billing_contract = BillingContract.find_by(id: billing_contract_id)
```

Comment:
doesnt self object already have biilling contract as association?
seems like you shouldnt make this call to DB

Reply:
done